Automattic/simplenote-macos
# Update Toolbar to Latest Design

I'm starting to update the app to the refreshed design set forth by @drw158. This PR mainly modifies the toolbar layout and some of its behaviors. And at last we get the search box in the top where it belongs :)

<img width="905" alt="screen shot 2017-11-10 at 2 41 17 pm" src="https://user-images.githubusercontent.com/789137/32682380-2fc0d26e-c629-11e7-8f3f-4f906e06d04c.png">

Most of the changes are new icons and XIB changes to outlets.

**To Test**
Launch the app, and try out all buttons in the toolbar. Expand the notes list size and app size to ensure resizing works as expected. Try out the new trash/restore behavior.

There's already one bug with this, the search field no longer styles itself properly when you change themes.

I'll send over some additional PRs for color and font updates.
